qwerty_ast/src/ast.rs

```diff
@@ -127,19 +145,19 @@ impl ASTNode {
     // Cringe helper function
     pub fn get_typ(&self) -> Option<Type> {
         match self {
-            ASTNode::QubitSymbol(val) => val.typ.clone(),
+            ASTNode::ZeroQubit(val) => val.typ.clone(),
             ASTNode::Tensor(val) => val.typ.clone(),
-            ASTNode::QubitLiteral(val) => val.typ.clone(),
+            ASTNode::OneQubit(val) => val.typ.clone(),
         }
     }
 
     // TODO: Fix this, not 100% sure it's right
     #[allow(unused)]
     pub fn get_dim(&self) -> Option<DimExpr> {
         match self {
-            ASTNode::QubitSymbol(_) => None,
+            ASTNode::ZeroQubit(_) => None,
             ASTNode::Tensor(_) => None,
-            ASTNode::QubitLiteral(val) => Some(val.factor.clone()),
+            ASTNode::OneQubit(val) => Some(val.factor.clone()),
         }
     }
 }
```

Comment:
It's not letting me comment on the lines below, but you can remove `factor` and `typ` from the `Tensor` struct